src/main/java/com/meta/cp4m/message/Message.java

```diff
@@ -32,13 +37,22 @@ static Identifier threadId(Identifier id1, Identifier id2) {
 
   Role role();
 
+  Message parentMessage();
+
   default Identifier threadId() {
     return threadId(senderId(), recipientId());
   }
 
   enum Role {
-    ASSISTANT,
-    USER,
-    SYSTEM
+    ASSISTANT(0),
+    USER(1),
+    SYSTEM(2);
+
+    public final Integer priority;
```

Comment:
So go ahead and get a copy of "Effective Java" by Joshua Bloch, it's a really great book for the foundation and sets a lot of Java conventions. 

"Item 16: In public classes, use accessor methods, not public fields"
"Item 61: Prefer primitive types to boxed primitives"
```suggestion
    private final int priority;
    
    public int priority() {
      return priority;
    }
```

Reply:
Addressed this!